src/library_lz4.js

```diff
@@ -42,6 +42,26 @@ mergeInto(LibraryManager.library, {
           end: file.end,
         });
       });
+      if (preloadPlugin) {
+        Browser.init();
+        pack['metadata'].files.forEach(function(file) {
+          var handled = false;
+          var fullname = file.filename;
+          Module['preloadPlugins'].forEach(function(plugin) {
+            if (handled) return;
+            if (plugin['canHandle'](fullname)) {
+              var dep = getUniqueRunDependency('fp ' + fullname);
+              addRunDependency(dep);
+              var finish = function() {
+                removeRunDependency(dep);
+              }
```

Comment:
Can this call `createPreloadedFile` the way normal files do? Or is the issue that we only uncompress the data if we find a plugin is relevant? If so I wonder if there isn't a way to refactor this to remove the redundancy, like giving `createPreloadedFile` a callback to get the data?

Regardless a comment around like 45 would be good to explain what this does and why it is here.